src/appleseed/foundation/math/beziercurve.h

```diff
@@ -351,6 +449,43 @@ inline typename BezierCurveBase<T, N>::VectorType BezierCurveBase<T, N>::transfo
     return VectorType(xpt.x * rcp_w, xpt.y * rcp_w, xpt.z * rcp_w);
 }
 
+
+template <typename T, size_t N>
+inline void BezierCurveBase<T, N>::transform_basis(const MatrixType& xfm)
+{
+    Matrix<ValueType, 4, 3> m_points = Matrix<ValueType, 4, 3>();
+    Matrix<ValueType, 4, 3> m_colors = Matrix<ValueType, 4, 3>();
+    Matrix<ValueType, 4, 1> m_widths = Matrix<ValueType, 4, 1>();
+    Matrix<ValueType, 4, 1> m_opacities = Matrix<ValueType, 4, 1>();
+
+    for (int32 i = 0; i < 4; ++i)
+    {
+        m_points[3 * i] = m_ctrl_pts[i].x;
+        m_points[3 * i + 1] = m_ctrl_pts[i].y;
+        m_points[3 * i + 2] = m_ctrl_pts[i].z;
+
+        m_colors[3 * i] = m_color[i].r;
+        m_colors[3 * i + 1] = m_color[i].g;
+        m_colors[3 * i + 2] = m_color[i].b;
+
+        m_widths[i] = m_width[i];
+        m_opacities[i] = m_opacity[i];
+    }
+
+    Matrix<ValueType, 4, 3> t_points = xfm * m_points;
```

Comment:
The meaning of the `t_` prefix isn't clear. Temp? In that case I suggest `temp_`.